[jdks] Fix connectTimeout for jdk download

`setup-disco` goal is broken in v1.24.0

It seems that the underlying plugin (`io.github.download-maven-plugin:download-maven-plugin:2.1.0`) doesn't support `connectTimeout` parameter.

```
[ERROR] Failed to execute goal org.jreleaser:jdks-maven-plugin:1.24.0:setup-disco (download-jre) on project jdemetra-app-snapshot: Unable to execute mojo: Unable to parse configuration of mojo io.github.download-maven-plugin:download-maven-plugin:2.1.0:wget for parameter connectTimeOut: Cannot find 'connectTimeOut' in class io.github.download.maven.plugin.internal.WGetMojo -> [Help 1]
```

Bug introduced in https://github.com/jreleaser/jreleaser/pull/2087/changes#diff-035349c56da14714b170ca52d06d3123014053672f734369120f70dbb264e296 for the issue https://github.com/jreleaser/jreleaser/issues/2083

Related code in `download-maven-plugin`: https://github.com/download-maven-plugin/download-maven-plugin/blob/e8c1b68c4bd5338fab1a77b9ca56430de8b498c6/src/main/java/io/github/download/maven/plugin/internal/WGetMojo.java#L807

--- a/plugins/jdks-maven-plugin/src/main/java/org/jreleaser/jdks/maven/plugin/JdkHelper.java
+++ b/plugins/jdks-maven-plugin/src/main/java/org/jreleaser/jdks/maven/plugin/JdkHelper.java
@@ -124,8 +124,11 @@ public class JdkHelper {
                     element("outputDirectory", jdkExtractDirectory.getAbsolutePath()),
                     element("cacheDirectory", cacheDirectory),
                     element("outputFileName", filename),
-                    element("connectTimeOut", String.valueOf(connectTimeout * 1000)),
-                    element("readTimeOut", String.valueOf(readTimeout * 1000))
+                    // download-maven-plugin 2.1.0 does not expose a separate connect timeout
+                    // parameter; WGetMojo wires readTimeOut into both connect and socket
+                    // timeouts internally. Pass the larger of the two so neither caller-set
+                    // value is silently dropped.
+                    element("readTimeOut", String.valueOf(Math.max(connectTimeout, readTimeout) * 1000))
                 ),
                 executionEnvironment(
                     project,

